lib/src/widgets/yust_file_picker.dart

```diff
@@ -249,17 +253,26 @@ class YustFilePickerState extends State<YustFilePicker> {
   }
 
   Widget _buildAddButton(BuildContext context) {
-    if (!_enabled) {
+    final canAddMore = widget.numberOfFiles != null
+        ? widget.files.length < widget.numberOfFiles! ||
+            (widget.numberOfFiles == 1 && widget.overwriteSingleFile)
+        : true;
+
+    if (!_enabled && !widget.overwriteSingleFile) {
+      return const SizedBox.shrink();
+    }
+    if (canAddMore) {
+      return IconButton(
+        iconSize: 40,
+        color: Theme.of(context).colorScheme.primary,
+        icon: const Icon(Icons.add_circle),
+        onPressed: _enabled && (widget.allowedExtensions?.isNotEmpty ?? true)
+            ? _pickFiles
+            : null,
+      );
+    } else {
```

Comment:
den `else`-Teil brauchst du nicht

Reply:
VS Code beschwert sich sonst aber:
"The body might complete normally, causing 'null' to be returned, but the return type, 'Widget', is a potentially non-nullable type.
Try adding either a return or a throw statement at the end."

Reply:
okay, dann nimm die Zeilen 262–264 gerne raus und ändere Z265 in `if (_enabled && canAddMore)` und return am Ende eine SizedBox:

```
if (_enabled && canAddMore) {
       return IconButton(
         iconSize: 40,
         color: Theme.of(context).colorScheme.primary,
         icon: const Icon(Icons.add_circle),
         onPressed: _enabled && (widget.allowedExtensions?.isNotEmpty ?? true)
             ? _pickFiles
             : null,
       );
     }
     
return const SizedBox.shrink();
```